### PR TITLE
Add configuration options

### PR DESCRIPTION
**Pull request by simon111111**
- Simon T's complete first version of gaplint README.md. Ready for notes from JDM.
- gaplint.py revision based on JDM June 14 notes complete. Ready for further notes from JDM if any. Primary revisions were a function that searches upwards through a directory tree searching for the config yml, and the resolution of a configuration precedence hierarchy.

### CODE REVIEW

**Comment by james-d-mitchell** [comment]
Can you please make the description and title a bit more descriptive?

**Comment by simon111111** [comment]
Sure, will do.On 14 Jun 2017 12:33 p.m., James Mitchell <notifications@github.com> wrote:Can you please make the description and title a bit more descriptive?

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/james-d-mitchell/gaplint","title":"james-d-mitchell/gaplint","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/james-d-mitchell/gaplint"}},"updates":{"snippets":[{"icon":"PERSON","message":"@james-d-mitchell in #4: Can you please make the description and title a bit more descriptive? "}],"action":{"name":"View Pull Request","url":"https://github.com/james-d-mitchell/gaplint/pull/4#issuecomment-308404500"}}}

File: gaplint.py [hunk 1]
```diff
@@ -695,56 +757,270 @@ def _parse_args(kwargs):
...
+    global _RULE_CODES
+    pattern1 = re.compile('\s*#')
+    pattern2 = re.compile('\s*$')
+    pattern3 = re.compile('\s*#\s*gaplint:\s*disable\s*=\s*')
+    global_supps = []
+    i = 0
+    n = len(lines)
+    myline = lines[0]
+    while  re.match(pattern1, myline) or re.match(pattern2, myline):
+        match = re.search(pattern3, myline)
```
Comment (james-d-mitchell) [inline gaplint.py]: I don't really understand the logic of this, if a line matches any number of whitespace followed by a `#`, then we look for `pattern3`?  Why is this `while` loop required at all?

File: gaplint.py [hunk 2]
```diff
@@ -695,56 +757,270 @@ def _parse_args(kwargs):
...
+                            return make_True_dic(_RULE_CODES)
+                        code = rule_code(rule)
+                        if not code in global_supps:
+                            global_supps.append(code)        
+        if i < n - 1:
+            i += 1
+            myline = lines[i]
+    return make_True_dic(global_supps)
+
+def add_global_supps_to_line_supps(line_supp_dic, global_supp_dic):
```
Comment (james-d-mitchell) [inline gaplint.py]: I'm not sure why you'd want to do this. This involves keeping a copy of the global suppressions for a file for every single line in the file, when only one copy is good enough. I think a different approach is required here.

File: gaplint.py [hunk 3]
```diff
@@ -695,56 +757,270 @@ def _parse_args(kwargs):
...
+        if not g_supp in l_supps:
+            line_supp_dic[g_supp] = True
+
+def get_line_suppressions(fname, line, linenum):
+    '''
+    Takes a filename, string and line number. Returns a dictionary whose keys
+    are the suppressed rules for the line, all assigned value True, and a 
+    boolean value True if the rules are to be applied to the next line and
+    False if not. If the suppression keywords are present, but no/invalid rules 
+    follow, a warning is thrown. A list is returned. The first element of which
```
Comment (james-d-mitchell) [inline gaplint.py]: Same comment as before, what if the keyword is correct but the value is wrong?

**Comment by simon111111** [comment]
Thanks a lot for the comments James. Will get on it and chat when you have
a chance today.

**Pull request closed without merging**